src/uct/ugni/rdma/ugni_rdma_ep.c

```diff
@@ -329,12 +330,128 @@ ucs_status_t uct_ugni_ep_atomic_cswap32(uct_ep_h tl_ep, uint32_t compare, uint32
     return UCS_ERR_UNSUPPORTED;
 }
 
```

Comment:
@hppritcha , I implemented both runtime and compile time check (is this a bit paranoiac?).
 
Based on the discovered device (https://github.com/shamisp/ucx/blob/topic/aries/src/uct/ugni/base/ugni_device.c#L91) we set the flag for the supported operations (https://github.com/shamisp/ucx/blob/topic/aries/src/uct/ugni/rdma/ugni_rdma_iface.c#L60).

The function pointers are always set, while in case of an old driver these return "unsupported" error code and  with the new driver we do this selection in a runtime + set the capability flags.